.github/workflows/api-ownership-updates.yml

```diff
@@ -0,0 +1,43 @@
+name: api-ownership-updates
```

Comment:
this shouldn't (?) be a separate workflow file or we can't make it required

Reply:
There are two other workflows that I thought adding it too:
*  openapi.yml: Seems related because they both are doing follow-ups for the same command `make build apis` but that one is triggered on push 
* openapi-diff.yml: That one's better because it's on pull request and not required already. But it's checking out a different repo so @mattgauntseo-sentry suggested defining this as a new workflow. 

I'm fine if this is not required but still at least run on every PR. It's not something that would fail so people won't skip it.